confdgnmi/src/confd_gnmi_server.py

```diff
@@ -216,12 +216,11 @@ def serve(port=PORT, adapter_type=AdapterType.DEMO, insecure=False,
             server.add_insecure_port("[::]:{}".format(port))
         else:
             assert key_file!= None and crt_file != None
-            with open(key_file) as k:
+            with open(key_file) as k, open(crt_file) as c:
                 key = k.read().encode("utf-8")
-                with open(crt_file) as c:
-                    crt = c.read().encode("utf-8")
-                    server.add_secure_port("[::]:{}".format(port),
-                                   grpc.ssl_server_credentials([(key, crt)]))
+                crt = c.read().encode("utf-8")
```

Comment:
One more small thing: you are opening the files in the text mode and then encoding the result back to bytes - just open them in `rb` mode and no encoding is necessary.